### DISCUSSION TITLE
Screen frozen and browser memory leak on select value in an option picker

### DISCUSSION THREAD

**Discussion by NicolasFabre81** [Budibase/budibase#11428] [Help] [answered]
Hello,

On a quite complex screen, I'm having trouble since yesterday : when I try to use this option picker : the screen freezes and the browser memory usage increase up to 5GB. The only solution is then to completely close the browser.
In the editor, the option picker works, no problem.

Of course I already try all what I could imagine :
_ delete the option picker and the related data provider and re-create it
_ change the name
_ check everywhere this field is in use 
etc ...
Sometimes it seems to solve the situation, but a bit after problem appears again, sometimes causing the entire screen to not load at all (we see only the spinner).

It seems like sth is "broken" in the page ...

Any help appreciated ! This screen required several days of work :-o
<img width="578" alt="image" src="https://github.com/Budibase/budibase/assets/28592069/e9b2942b-15df-41c2-9eec-5183ad91b71a">
<img width="350" alt="image" src="https://github.com/Budibase/budibase/assets/28592069/d61cf312-0d5d-49f4-ab48-5ad4017eb0b0">

**Comment by NicolasFabre81** [comment]
When the entire screen does not load, it is caused to 2 data providers (those of the 2 option pickers "commandes / clients" and "article" ) that do not respond :
<img width="595" alt="image" src="https://github.com/Budibase/budibase/assets/28592069/8d0724fd-f375-46f6-94d3-6931f00c54b5">

But queries are OK, what could cause this ?

**Comment by melohagan** [comment]
Hey @NicolasFabre81 

This looks to be quite a complex app! What potentially could be happening here is that you have many Data providers which have filters against some bindings?

It is worth noting that when a binding value is updated, it will refresh (re-fetch data) for any data providers that link to that bindings.

You options picker could possibly be triggering a cascade of fetching, maybe even in a loop, and that could be causing issues. This is just a guess though.

**Comment by NicolasFabre81** [comment, marked as answer]
Hello Melohagan,
Tanks for the reply, I finally solved it by restoring a backup, then
renamed the option picker field and checked all the dependencies.
Anyway your hints are precious, good to understand how it works.
And yes this is a so complex screen with à lot of DP, conditions rules and
big sql queries !
Unfortunately I must leave it for a month : holyday time ! But I confess
I'll miss my bb toy...
